marketplace/netlify-build/test/index.spec.js

```diff
@@ -0,0 +1,232 @@
+import React from 'react';
+import { render, cleanup, wait, fireEvent } from '@testing-library/react';
+import NetlifyExtension from '../src/index.js';
+import { createPubSub } from '../src/pubnub-client';
+
+const mockaddListener = jest.fn();
+const mocksubscribe = jest.fn();
+const mockpublish = jest.fn();
+const mockhistory = jest.fn(() =>
+  Promise.resolve({
+    messages: [
+      {
+        timetoken: '15620690994438613',
+        entry: {
+          contentful: true,
+          event: 'triggered',
+          userId: '2jvc3kU4n7OIABiFMTaGyB'
+        }
+      }
+    ],
+    startTimeToken: '15620690994438613',
+    endTimeToken: '15620690994438613'
+  })
+);
+
+jest.mock('pubnub', () => {
+  return {
+    __esModule: true,
+    default: jest.fn(() => ({
+      addListener: mockaddListener,
+      subscribe: mocksubscribe,
+      publish: mockpublish,
+      history: mockhistory,
+      removeListener: jest.fn(),
+      unsubscribe: jest.fn()
+    }))
+  };
+});
+
+
+// date-fns provides functions such as `distanceInWordsToNow` which will change as the time changes
+// it is necessary to ensure the date will remain the same no matter when these tests are run
+global.Date.now = jest.fn(() => 1562082624100);
+
+const sdkMock = {
+  parameters: {
+    instance: {},
+    installation: {
+      names: 'upbeat,friendly',
+      channels:
+        'c-wj2QelNg1OvkkWDF-5133ac93-7fc0-463b-96d0-bb799e7fcb07,c-wj2QelNg1OvkkWDF-119493e8-f2a3-4771-8b3c-f402b5966e84',
+      publishKey: 'pub-c-a99421b9-4f21-467b-ac0c-d0292824e8e1',
+      subscribeKey: 'sub-c-3992b1ae-f7c9-11e8-adf7-5a5b31762c0f',
+      buildHookUrls:
+        'https://api.netlify.com/build_hooks/5d19c2c1b359da07eca189af,https://api.netlify.com/build_hooks/5d19c2c1b359da2f7da18998',
+      netlifySiteIds: '5133ac93-7fc0-463b-96d0-bb799e7fcb07,119493e8-f2a3-4771-8b3c-f402b5966e84',
```

Comment:
Are these real? We can keep them but make sure you deleted Netlify sites.

Reply:
They are real and I own them. They will be deleted from my personal Netlify once I'm done so no worries there.